### PR TITLE
feat: Add docker config

### LINKED ISSUE #122: Build docker images
- Setup a Dockerfile according to Dockerflow guidelines to build a syncstorage-rs image
- Setup circleci to build it

See autopush-rs's or channelserver's Dockerfile and the [Dockerflow repo's](https://github.com/mozilla-services/Dockerflow) examples for good bases to start from.

### PR DESCRIPTION
**Pull request by jrconlin**
WIP

Closes #122

NOTES:
* Probably want to run `docker-compose up` on this to get both the dummy database and app running
* ~App is complaining that "System is not running". I'm guessing that this is a different error, but will investigate more.~
* Added note about making damn sure that `actix-web` and `actix` versions are compatible when updating. This means no dependbot or auto-updates for a while... 😭

### CODE REVIEW

File: src/server/mod.rs [hunk 1]
```diff
@@ -137,7 +137,7 @@ impl Server {
 
             vec![build_app(state), build_dockerflow(dfstate)]
         })
-        .bind(format!("127.0.0.1:{}", settings.port))
+        .bind(format!("0.0.0.0:{}", settings.port))
```
Comment (pjenvey) [inline src/server/mod.rs]: is the new settings.host actually used anywhere (looks like it should be here)?

Comment (jrconlin) [reply src/server/mod.rs]: oh for... forgot to commit that change.

File: src/server/mod.rs [hunk 2]
```diff
@@ -137,7 +137,7 @@ impl Server {
 
             vec![build_app(state), build_dockerflow(dfstate)]
         })
-        .bind(format!("127.0.0.1:{}", settings.port))
+        .bind(format!("{}:{}", settings.host, settings.port))
```
Comment (pjenvey) [inline src/server/mod.rs]: FYI `.bind((settings.host.as_ref(), settings.port))` also works, there's a ToSocketAddr's impl for that kind of 2 item tuple

Comment (jrconlin) [reply src/server/mod.rs]: Yeah, wasn't sure if "127.0.0.1" was something explicitly set, otherwise would have done something like that.

**Pull request merged**